### ISSUE TITLE
Consolidate code from pages

### ISSUE THREAD

**Issue by Jason9199** [FreeCodeCampOKC/fccokc_web_legacy#46]
Consolidate website into views (probably using Angular) to eliminate repetition of code across the pages. Then configure routing to display the appropriate view of the body based on the route of the URL.

Suggested approach:
- create a master layout view that references all the partials
- create view for the header
- create view for the footer
- reduce the pages to views of the body of the page without the header or footer included in the view

**Comment by mattstub** [comment]
I like the idea of this because there is a lot of repetitive code, just wondering how difficult it may be for some people who do not have experience with Angular, to contribute and navigate the project?

**Comment by Jason9199** [comment]
Correct. And because of that it should have an Intermediate or Advanced label on it. I'm happy to handle it myself, but I haven't had time to start on it yet.

**Comment by kacollins** [comment]
Let's put this one on hold until after Hacktoberfest :)

**Comment by CarsonMcKinstry** [comment]
While this is on hold, I do want to comment on this!

So, I think we ought to use something that isn't Angular for this? Angular is rather large and causes slowdowns on initial loads, even if the original file is minified. It just feels a bit heavy-handed.

The point of GH pages is to use static HTML files or a generator like Jekyll. As well, GH pages, unfortunately, doesn't support angular routing, which is required for a SPA.

**Comment by Jason9199** [comment]
I hadn't heard of any issues with Angular routing and GitHub pages, that would be a shame. But I don't know if I've ever heard of Angular described as heavy or slow. I've used it a lot and never experienced those symptoms.

If we decide to do something other than Angular, we should still be able to come up with a solution that doesn't have so much duplicated code.

**Comment by CarsonMcKinstry** [comment]
Back when I first started trying to use Angular for my own site on GH pages, routing didn't work and I found that angular-router and ui-router both required a web server for the routes generated by either of those libraries to work. The solution is creating a flat file, with every single page in a single html file.  That's not good or best practice.

Minified, Angular is 143k, and since we don't have a way to gzip files, that's the best we can do. For a static site, that is humongous and a tad bit unnecessary, especially since we don't have a lot of dynamic content.

One solution would be to go to a hosted solution such as Heroku. It would allow us to have much more flexibility, and we could also introduce people to server-rendered sites with something like ejs.
